akka-http-core/src/main/java/akka/http/javadsl/model/headers/ModeledCustomHeader.java

```diff
@@ -0,0 +1,43 @@
+package akka.http.javadsl.model.headers;
+
+import akka.http.impl.util.Rendering;
+import akka.util.Helpers;
+
+/**
+ * Support class for building user-defined custom headers defined by implementing `name` and `value`.
+ * By implementing a {@link ModeledCustomHeader} along with {@link ModeledCustomHeaderCompanion} instead of {@link CustomHeader} directly,
+ * convenience methods that allow parsing this user-defined header from {@link akka.http.javadsl.model.HttpHeader}  are
+ * available to use.
+ */
+public abstract class ModeledCustomHeader extends CustomHeader {
```

Comment:
That I don't know, I was just emulating the Scala API, maybe somebody from the team can explain.